### PR TITLE
Allow calling of `sendMessage` with props OR text

### PR DESCRIPTION
**Pull request by jugarrit**

### CODE REVIEW

File: test/specs/services/conversation.spec.js [hunk 1]
```diff
@@ -240,7 +240,20 @@ describe('Conversation service', () => {
 
         conversationStartedSuite(conversationService.sendMessage('message'));
 
-        it('should add message and send message', () => {
+        it('should add message and send message', function*() {
+            mockedStore = createMockedStore(sandbox, getProps());
+
+            yield mockedStore.dispatch(conversationService.sendMessage('message'));
```
Comment (lemieux) [inline test/specs/services/conversation.spec.js]: should test the outcome of this though

File: test/specs/services/conversation.spec.js [hunk 2]
```diff
@@ -240,7 +240,20 @@ describe('Conversation service', () => {
...
+            yield mockedStore.dispatch(conversationService.sendMessage({
+                type: 'text',
+                'text': 'message'
```
Comment (lemieux) [inline test/specs/services/conversation.spec.js]: no need for `' '` around `text` here

File: test/specs/services/conversation.spec.js (hunk 1 above)
Comment (jugarrit) [inline test/specs/services/conversation.spec.js]: We are? We're checking that `postSendMessage` is called - one for each message.
Unless you mean check the args, too? I can do that.

Comment (lemieux) [inline test/specs/services/conversation.spec.js]: Yeah I mean, does it transfer that string arg to message.text properly.

File: src/js/services/conversation.js [hunk 3]
```diff
@@ -91,12 +91,13 @@ const addMessage = (text, extra = {}) => {
...
-            ...extra
+            sendStatus: SEND_STATUS.SENDING
         };
 
-        if (text) {
-            message.text = text;
+        if (typeof props === 'string') {
+            message.text = props;
+        } else {
+            Object.assign(message, ...props);
```
Comment (mspensieri) [inline src/js/services/conversation.js]: How is the spread working here? Shouldn't this be either

```
Object.assign(message, props)
```

or

```
message = {
   ...message,
   ...props
}
```

?

Comment (jugarrit) [inline src/js/services/conversation.js]: I don't know why it's working @mspensieri but I'll change it so that it makes more sense to us

Comment (lemieux) [inline src/js/services/conversation.js]: I guess it spreads the object as multiple objects with only one key and one value and `Object.assign` accepts multiple objects in a row

**Pull request merged**